components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -125,17 +138,67 @@ void Nudging::initialize_impl (const RunType /* run_type */)
+  // Initialize the refining remapper stuff at the outset,
+  // because we need to know the grid information
+  if(m_refine_remap) {
+    // For now, we are doing the horizontal interpolation last,
+    // so we use the m_grid (model physics) as the target
+    // TODO: maybe clean this up?
+    auto grid_tgt = m_grid->clone(m_grid->name(), false);
+    auto refine_remapper_p2p =
+        std::make_shared<RefiningRemapperP2P>(grid_tgt, m_refine_remap_file);
+    refine_remapper = refine_remapper_p2p;
+  }
+
+  // Set the external grids
+  // We have three types of grids with different behaviors
+  // TODO: definitely clean this up
+
+  // The grid from the remapper needs a const AbstractGrid, but
+  // the other two grids need a non-const AbstractGrid
+  // TODO: What is actually going on here anyway?
+  std::shared_ptr<const scream::AbstractGrid> grid_ext_const;
+  std::shared_ptr<scream::AbstractGrid> grid_ext;
+  std::shared_ptr<scream::AbstractGrid> grid_hxt;
+
+  if(m_refine_remap) {
+    // If we are refine-remapping, then get grid from remapper
+    grid_ext_const = refine_remapper->get_src_grid();
+    // Deep clone it though to get rid of "const" stuff
+    grid_ext = grid_ext_const->clone(grid_ext_const->name(), false);
+  } else {
+    // If not refine-remapping, then use whatever was used before,
+    // i.e., deep clone the physics grid
+    grid_ext = m_grid->clone(m_grid->name(), false);
+  }
+
+  // The ultimate grid is grid_ext (external grid, i.e., files),
+  // so, grid_ext can potentially have different levels
   grid_ext->reset_num_vertical_lev(m_num_src_levs);
+  // Declare the layouts for the helper fields (ext --> mid)
   FieldLayout scalar2d_layout_mid { {LEV}, {m_num_src_levs} };
   FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols, m_num_src_levs} };
+  // The penultimate grid is grid_hxt (external horiz grid, but model physics
+  // vert grid, so potentially a bit of a mess)
+  grid_hxt = grid_ext->clone(grid_ext->name(), false);
+  auto h_num_levs = m_num_levs;
+  grid_hxt->reset_num_vertical_lev(h_num_levs);
+  // Declare the layouts for the helper fields (hxt --> hid)
+  // TODO: use better names
+  FieldLayout scalar2d_layout_hid { {LEV}, {h_num_levs}};
+  FieldLayout scalar3d_layout_hid { {COL,LEV}, {m_num_cols, h_num_levs} };
+
+  // Note: below, we only need to deal with the pressure stuff on ext_grid, not
+  // hxt_grid because we are not doing vertical interpolation on the hxt_grid
+
+  // Initialize the time interpolator
   m_time_interp = util::TimeInterpolation(grid_ext, m_datafiles);
 
   constexpr int ps = SCREAM_PACK_SIZE;
-  const auto& grid_name = m_grid->name();
+  // To be extra careful, this should be the ext_grid
+  const auto& grid_ext_name = grid_ext->name();
   if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
-    create_helper_field("p_mid_ext", scalar3d_layout_mid, grid_name, ps);
+    create_helper_field("p_mid_ext", scalar3d_layout_mid, grid_ext_name, ps);
```

Comment:
I noticed you have a few places where you create a helper field, and immediately retrieve it. You may consider changing `create_helper_field` to return the newly created field, to make the code simpler.

Instead of
```
create_helper_field("blah",...);
auto blah = get_helper_field("blah");
```
simply do
```
auto blah = create_helper_field("blah");
```